### DISCUSSION TITLE
Build and View file with fig.eps

### DISCUSSION THREAD

**Discussion by saulormedrado** [texstudio-org/texstudio#3112] [General]
Hi everyone. I made some epslatex figures on gnuplot, but I had some problems when I use it on Texstudio.  I only got  the pdf document if i put Options > Build > Build & View : DVI->PS->PDF Chain. But in this mode, I lose the feature that allow to go code if i click with right botton in  pdf document. If I put  Compile & View, i got the error of compiller and o dont obtain a document pdf. What I did wrong?

**Comment by sunderme** [comment]
pdflatex does not understand eps. I would suggest png images instead.

**Comment by muzimuzhi** [comment]
> if i click with right botton in pdf document. If I put Compile & View, i got the error of compiller and o dont obtain a document pdf.

What's the error message?

> pdflatex does not understand eps.

The `\includegraphics` from `graphicx` package supports inclusion of eps images by calling `repstopdf` ("r" stands for restricted), a Ghostscript wrapper in Perl to first convert an eps to pdf, then include the resulted pdf.  (No `-shell-escape` needed because `repstopdf` is one of the allowed (safe) executable preconfigured in `texmf`.)

Suppose you have an `example-eps-image.eps`, then running pdflatex on the example below will first convert it to `example-eps-image-eps-converted-to.pdf` then include the new pdf.  The whole process is automatic.  Note the file extension `.eps` should always be omitted.  You'll need both [`epstopdf`](https://ctan.org/pkg/epstopdf) (provides Perl script) and [`epstopdf-pkg`](https://ctan.org/pkg/epstopdf-pkg) (provides LaTeX interface) packages.
```tex
\documentclass{article}
\usepackage{graphicx}

\begin{document}
\includegraphics{example-eps-image}
\end{document}
```

**Reply by saulormedrado** [reply]
Thanks for answering. I made a simple code and it worked. The problem is that my figure is in another folder. But I have a lot of figures, so it's better for me that the eps is in another folder. What do I do?
The error is " Package pdftex.def Error: File `../Plot/example-eps-image-eps-converted-to.pdf' not found: using draft setting. ...]{../Plot/example- eps-image} " and this path is right.

**Reply by muzimuzhi** [reply]
Turns out it's a LaTeX usage question which is irrelevant to the editor in use.  Mind posting to an online Q&A forum such as tex.stackexchange.com?

**Reply by muzimuzhi** [reply]
> But I have a lot of figures, so it's better for me that the eps is in another folder. What do I do?

In a simpler case that images are located in sub-dir(s), setting `\graphicspath{{sub-dir-a/}{sub-dir-b/}}` should work.

But in your case, images are located in a sibling dir, where the restricted `repstopdf` refuse to touch.  In my opinion the expected behavior is to use `-shell-escape` and then `graphicx` calls the non-restricted `epstopdf` script.  Before https://github.com/ho-tex/epstopdf/issues/3 is resolved, you can achieve the expected behavior by loading `pdftexcmds` before `graphicx`:

```tex
\documentclass{article}
\usepackage{pdftexcmds}
\usepackage{graphicx}
\graphicspath{{../Plot}}

\begin{document}
% provided there exists "../Plot/example-eps-image.eps"
\includegraphics{example-eps-image}
\end{document}
```

**Reply by saulormedrado** [reply]
I understand. Thank you for help me.
